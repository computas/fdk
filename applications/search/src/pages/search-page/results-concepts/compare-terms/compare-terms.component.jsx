import React from 'react';
import PropTypes from 'prop-types';

import localization from '../../../../lib/localization';
import { getTranslateText } from '../../../../lib/translateText';

export const CompareTerms = props => {
  const { prefLabel, creator, onDeleteTerm, termIndex } = props;
  let title = getTranslateText(prefLabel);
  title = title.charAt(0).toUpperCase() + title.substring(1).toLowerCase();
  return (
    <div className="fdk-container p-3">
      <button
        className="float-right fdk-button-small"
        onClick={() => {
          onDeleteTerm(termIndex);
        }}
      >
        <i className="fa fa-times" />
        &nbsp;
        {localization.terms.removeTerms}
      </button>
      <h4 className="clearfix">{title}</h4>
      {creator && <div>{creator}</div>}
    </div>
  );
};

CompareTerms.defaultProps = {
  prefLabel: null,
  creator: null
};

CompareTerms.propTypes = {
  prefLabel: PropTypes.object,
  creator: PropTypes.string,
  onDeleteTerm: PropTypes.func.isRequired,
  termIndex: PropTypes.number.isRequired
};
