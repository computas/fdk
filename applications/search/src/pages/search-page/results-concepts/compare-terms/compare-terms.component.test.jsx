import React from 'react';
import { shallow } from 'enzyme';
import { CompareTerms } from './compare-terms.component';

let defaultProps;
let wrapper;
let onDeleteTerm;

beforeEach(() => {
  onDeleteTerm = jest.fn();
  defaultProps = {
    prefLabel: {
      nb: 'PrefLabel 1'
    },
    creator: 'Creator 1',
    onDeleteTerm,
    termIndex: 0
  };
  wrapper = shallow(<CompareTerms {...defaultProps} />);
});

test('should render CompareTerms correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onDeleteTerm', () => {
  wrapper.find('button').prop('onClick')(defaultProps.termIndex);
  expect(onDeleteTerm).toHaveBeenLastCalledWith(defaultProps.termIndex);
});
