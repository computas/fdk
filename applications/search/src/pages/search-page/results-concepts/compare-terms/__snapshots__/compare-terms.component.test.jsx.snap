// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CompareTerms correctly 1`] = `
<div
  className="fdk-container p-3"
>
  <button
    className="float-right fdk-button-small"
    onClick={[Function]}
  >
    <i
      className="fa fa-times"
    />
    Â 
    Remove
  </button>
  <h4
    className="clearfix"
  >
    Preflabel 1
  </h4>
  <div>
    Creator 1
  </div>
</div>
`;
